reframe/frontend/runreport.py

```diff
@@ -270,3 +265,94 @@ def junit_dump(xml, fp):
         etree.tostring(xml, encoding='utf8', pretty_print=True,
                        method='xml', xml_declaration=True).decode()
     )
+
+
+def get_reports_files(directory):
+    return [f for f in glob.glob(f"{directory}/*")
+            if os.path.isfile(f) and not f.endswith('/latest.json')]
+
+
+def index_report(report, filename):
+    prefix = os.path.dirname(filename)
+    index_file = os.path.join(prefix, 'index.db')
+    if not os.path.exists(index_file):
+        getlogger().info('Re-building the report index...')
+        _build_index(index_file)
+
+    with sqlite3.connect(index_file) as conn:
+        _index_report(conn, report, filename)
+
+
+def _build_index(filename):
+    with sqlite3.connect(filename) as conn:
+        conn.execute(
+'''CREATE TABLE IF NOT EXISTS testcases(
+        name TEXT,
+        hash TEXT,
+        system TEXT,
+        partition TEXT,
+        environment TEXT,
+        time_start TEXT,
+        time_end TEXT,
+        run_index INTEGER,
+        test_index INTEGER,
+        report_file TEXT
```

Comment:
Can we add more fields like the number of cases and number of failures?

Reply:
I would love to see also the user, version, and time_elapsed fields. But this would require another table. I think we should do two tables. One with the test cases and another one with the runs. I would love to make fast reports using the later. This would have the num_cases and num_failures in the table. What do you think?

Reply:
In the second table ("experiments") we could add the "runs" fields, num_cases, num_failures, num_aborted, and num_skipped. What do you think?